docs/src/main/sphinx/language/lexical-structure.rst

```diff
@@ -0,0 +1,135 @@
+=================
+Lexical structure
+=================
+
+A SQL statement consist of a series of tokens, separated by whitespace
+characters, and terminated by a semicolon.
+
+A token can one be any of the following:
+
+* A keyword
+* An identifier
+* A quoted identifier
+* A literal (constant)
+* A special character symbol
+
+Valid whitespace characters are space, tab, or a newline.
+
+Statements can also contain comments, but they are not tokens, so they are
+effectively ignored.
+
+The following example contains all elements::
+
+    SELECT * FROM example_table WHERE a_column = '5'; -- a comment
+
+* ``SELECT``, ``FROM``, and ``WHERE`` are tokens.
+* The ``=`` operator, the  ``*`` symbol and the semicolon ``;`` are all  special
+  characters.
+* Space acts as whitespace to separate elements.
+* ``example_table``, and ``a_column`` are identifiers.
+* ``--`` starts a comment section for the rest of the line.
+
+Keywords and identifiers
+------------------------
+
+Keyword tokens, such as ``SELECT``, have a fixed meaning in the SQL language.
+Tokens that identify names of tables, columns, functions, or other objects, are
+identifiers. Keywords and identifiers have the same lexical structure, meaning
+that one cannot know whether a token is an identifier or a key word without
+knowing the language.
+
+The :doc:`list of reserved keywords </language/reserved>` includes all tokens
+used for the supported :doc:`/sql`.
+
+Identifiers must start with a letter and subsequently include letters, digits,
+and underscores. A dot is used to compose multiple identifiers to a of a
+qualified identifier. Identifiers with special characters must be quoted.
```

Comment:
People asked a few times on Slack what characters are allowed in quoted identifiers, so it's worth explaining that anything is valid and only double quotes need to be escaped.
```suggestion
qualified identifier. Identifiers with special characters must be quoted.
A quoted identifier can contain any characters. If it includes a double quote
character, it needs to be escaped with another double quote character.
```